erpnext/accounts/report/trial_balance/trial_balance.py

```diff
@@ -87,8 +87,7 @@ def get_opening_balances(filters):
 
 
 def get_rootwise_opening_balances(filters, report_type):
-	additional_conditions = " and posting_date >= %(year_start_date)s" \
-		if report_type == "Profit and Loss" else ""
+	additional_conditions = ""
```

Comment:
removing those lines appears to fix unclosed opening balance for income +expense on trial balance sheet so they behave like liability and equity and assets